metagraph/src/graph/alignment/aligner_extender_methods.cpp

```diff
@@ -50,709 +55,695 @@ DefaultColumnExtender<NodeType>::DefaultColumnExtender(const DeBruijnGraph &grap
-    constexpr size_t column_vector_size = sizeof(std::pair<NodeType, std::pair<Column, bool>>);
-
-    auto get_column_size = [&](const Scores &scores) {
-        size_t size = std::get<0>(scores).capacity();
-        return sizeof(Scores) + size * (
-            sizeof(score_t) * 3 + sizeof(Cigar::Operator) * 3 + sizeof(NodeId) * 2
-        );
-    };
-    size_t total_size = column_vector_size + get_column_size(first_column);
+auto DefaultColumnExtender::extend(score_t min_path_score) -> std::vector<Alignment> {
+    assert(this->seed_);
 
-    S[0] = seed_->get_score() - profile_score_[seed_->get_sequence().back()][start + 1];
+    table.clear();
+    prev_starts.clear();
 
-    AlignNode start_node{ graph_.max_index() + 1,
-                          seed_->get_sequence()[seed_->get_sequence().size() - 2],
-                          0, 0 };
+    size_t start = this->seed_->get_clipping();
 
-    typedef std::pair<AlignNode, score_t> Ref;
-    Ref best_start{ start_node, S[0] };
-    std::vector<Ref> starts;
+    // the sequence to align (the suffix of the query starting from the seed)
+    std::string_view window(this->seed_->get_query().data(),
+                            query_.data() + query_.size() - this->seed_->get_query().data());
 
-    std::priority_queue<Ref, std::vector<Ref>, utils::LessSecond> stack;
-    stack.emplace(start_node, S[0]);
+    ssize_t seed_offset = static_cast<ssize_t>(this->seed_->get_offset()) - 1;
 
-    while (stack.size()) {
-        AlignNode prev = stack.top().first;
-        stack.pop();
+    // initialize the root of the tree
+    table.emplace_back(alloc_column<Column>(window.size() + 1, this->seed_->front(),
+                                            static_cast<size_t>(-1), '\0', seed_offset, 0, 0));
+    std::fill(std::get<0>(table[0]).begin(), std::get<0>(table[0]).end(), 0);
 
-        if (static_cast<double>(total_size) / 1000000
-                > config_.max_ram_per_alignment) {
-            DEBUG_LOG("Alignment RAM limit reached, stopping extension");
-            break;
-        }
-
-        if (static_cast<double>(num_columns) / extend_window_.size()
-                > config_.max_nodes_per_seq_char) {
-            DEBUG_LOG("Alignment node limit reached, stopping extension");
-            break;
-        }
+    score_t xdrop_cutoff = std::max(-config_.xdrop, ninf + 1);
+    assert(config_.xdrop > 0);
+    assert(xdrop_cutoff < 0);
 
-        size_t next_distance_from_origin = std::get<3>(prev) + 1;
+    using TableIt = std::tuple<score_t,
+                               ssize_t, /* negative off_diag */
+                               size_t /* table idx */>;
+    TableIt best_score { 0, 0, 0 };
 
-        for (const auto &[next, c] : get_outgoing(prev)) {
-            auto &column_pair = table_[next];
-            auto &[column, converged] = column_pair;
-            if (converged)
-                continue;
+    // node traversal heap
+    std::priority_queue<TableIt> queue;
```

Comment:
do you actually ever call `std::get<1>(...)` on these tuples?

Reply:
not really, they're just there for breaking ties

Reply:
Ok, please write this in a comment.